internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -133,6 +133,11 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		return ctrl.Result{}, errors.Wrap(err, "failed to apply patch label to the node")
 	}
 
+	// Reconcile node taints
+	if err := r.reconcileNodeTaints(ctx, remoteClient, node); err != nil {
```

Comment:
After the SSA caching PR we probably have to make sure that this func gets the updatedNode. Let's circle back once caching is merged

Reply:
I think with the latest implementation you have to pass updatedNode in here.

But please do a manual test that reconciles of the taints work in both cases (when labels/annotations are written and also when that was just a no-op)